src/blinding.cpp

```diff
@@ -298,6 +301,37 @@ std::pair<uc32, cleared_uc32> blind25_key_pair(
     return result;
 }
 
+static const auto version_blinding_hash_key_sig = to_unsigned_sv("VersionCheckKey_sig"sv);
+
+std::pair<uc32, cleared_uc64> blind_version_key_pair(ustring_view ed25519_sk) {
+    std::array<unsigned char, 64> ed_sk_tmp;
+    if (ed25519_sk.size() == 32) {
+        std::array<unsigned char, 32> pk_ignore;
+        crypto_sign_ed25519_seed_keypair(pk_ignore.data(), ed_sk_tmp.data(), ed25519_sk.data());
+        ed25519_sk = {ed_sk_tmp.data(), 64};
+    }
+    if (ed25519_sk.size() != 64)
+        throw std::invalid_argument{
+                "blind_version_key_pair: Invalid ed25519_sk is not the expected 32- or 64-byte "
+                "value"};
+
+    uc32 seed;
+    crypto_generichash_blake2b(
+            seed.data(),
+            seed.size(),
+            ed25519_sk.data(),
+            32,
+            version_blinding_hash_key_sig.data(),
+            version_blinding_hash_key_sig.size());
+
+    uc32 pk;
+    cleared_uc64 sk;
```

Comment:
```suggestion
    std::pair<uc32, cleared_uc64> result;
    auto& [pk, sk] = result;
```
and then just `return result;` at the end of the func.

By constructing directly into a value of the function's return type, and then returning that value, C++ return value optimization kicks in and will avoid having a that return variable in this function at all: the compiler will write the returned value directly into wherever the calling function is going to store it.

Reply:
(Included this change in an earlier suggestion, but leaving this here to explain why the change to a std::pair usage).

Reply:
I'm not sure how long it'll take me to get used to this `auto& [pk, sk] = result;` convention (it's just so different from most of the languages I normally work in 😅) but good to know that it also results in optimisations in the compiler when done

Have updated the code 🙏

Reply:
It's a shortcut for `auto& pk = result.first; auto& sk = result.second;`, but works for std::pairs, std::array, std::tuples, and simple structs.  It's particularly useful in looping through maps:
```C++
for (auto& [k, v] : somemap) { /*...*/ }
```

Reply:
Ah yea that makes a bit more sense in my head haha